index.js

```diff
@@ -27,16 +28,55 @@ const main = (fn, fp, data, opts) => {
 		});
 	}
 
-	const json = JSON.stringify(data, opts.replacer, opts.indent);
+	return fn(fp, data, opts);
+};
+
+const readFile = fp => {
+	return pify(fs.readFile)(fp, 'utf8')
+		.catch(err => {
+			if (err.code === 'ENOENT') {
+				return null;
+			}
+
+			throw err;
```

Comment:
Well, it fails silently if the file doesn't exist. But I guess we can safely swallow all errors regardless.